docs/design/Chem Integration.md

```diff
@@ -0,0 +1,66 @@
+# Chem integration into Upgrade Assistant
+
+Chem is a tool developed internally to solve the problem of matching loose assemblies (ie `.dll` files added into projects with unknown provenance). As customers are looking to move to .NET 5, this may pose a challenge as these versions are highly likely to be .NET Framework compilations that may not work as expected on newer platforms. Often, these libraries have NuGet packages now and newer versions that could work on newer platforms.
+
+The main idea behind Chem is based on the idea of chemical fingerprints (hence the naming) that some small amount of data may be used to infer the original identity. In the case of binaries, there are a number of fingerprints that may be useful in identifying what it may be:
+
+- File name
+- Public key token
+- File contents (and hashes thereof)
+- and others
+
+Work has already been done to generate a database of known packages along these and other identity dimensions. This data can then be used to generate potential indexes that could be used from within Upgrade Assistant to identify along this potential dimensions, both for notification and even to update a project file to reference the NuGet package rather than the loose assembly.
+
+## User Experience
+
+The main scenario this will focus on is one where a project has downloaded an assembly and added it to its repo and referenced it from various projects. The end result of this work should be able to:
+
+- Identify what package the dependency came from
+- Update a project file to now reference this from
```

Comment:
Nit-pick, something seems off about this sentence. Maybe a missing word or something? I think it's meant to say "Update a project file to now reference this package" or something along those lines.